### PR TITLE
docs: Support management port for health and metrics

### PR DESCRIPTION
**Pull request by mabartos**
Relates to #19334

### CODE REVIEW

File: docs/documentation/release_notes/topics/25_0_0.adoc [hunk 1]
```diff
@@ -33,4 +33,15 @@ For users of the `keycloak-authz-client` library, calling `AuthorizationResource
...
+= Management port for metrics and health endpoints
+
+Metrics and health checks endpoints are no longer accessible through the standard {project_name} server port.
+As these endpoints should be hidden from the outside world, they can be accessed on a separate default management port `9000`.
+
+It allows to not expose it to the users as standard Keycloak endpoints in Kubernetes environments.
+The new management interface provides a new set of options and is fully configurable.
+
+{project_name} Operator assumes the management interface is turned on by default.
+For more details, see https://www.keycloak.org/server/management-interface[Configuring {project_name} Management Interface].
```
Comment (mabartos) [inline docs/documentation/release_notes/topics/25_0_0.adoc:47]: I think tests around "external" links will fail due to this change, as the newly created guide is not publicly visible on the website yet.

@ahus1 What is the preferred approach how to reference the newly introduced guide?

**Comment by mabartos** [comment]
@ahus1 @vmuzikar @andymunro Docs for the management interface should be ready for review.

**Review by ahus1 — CHANGES_REQUESTED** [review]
@mabartos - thank you for these docs, they will help uses a lot. 

See https://github.com/mabartos/keycloak/pull/769 for a PR which beyond some smaller edits moves the Operator parts to the advanced operator stuff. 

Please let me know what you think. It might be that more changes are necessary after the move that I have missed.

File: docs/documentation/release_notes/topics/25_0_0.adoc (hunk 1 above)
Comment (ahus1) [reply docs/documentation/release_notes/topics/25_0_0.adoc:47]: The link looks fine. The external link check will fail, but it won't block us to merge the PR.

**Comment by mabartos** [comment]
@ahus1 These changes make sense to me. Thanks!

**Review by ahus1 — DISMISSED** [review]

**Comment by mabartos** [comment]
@vmuzikar PTAL, once you have time, thank you.

File: docs/guides/server/management-interface.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+The management interface is turned on by default, so management endpoints such as `/metrics`, and `/health` are exposed on the default management port `9000`.
+In order to change the port for the management interface, you can use the {project_name} option `http-management-port` or environment variable `KC_HTTP_MANAGEMENT_PORT`.
+
+The management interface provides a set of options and is fully configurable.
+If these options for the management HTTP server are not explicitly set, their values are automatically inherited from the default HTTP server.
+
+You can change the relative path of the management interface, as the prefix path for the management endpoints can be different.
+You can achieve it via the {project_name} option `http-management-relative-path` or environment variable `KC_HTTP_MANAGEMENT_RELATIVE_PATH`.
+
+For instance, if you set the CLI option `--http-management-relative-path=/management`, the metrics, and health endpoints will be accessed on the `/management/metrics` and `/management/health` paths.
```
Comment (thomasdarimont) [inline docs/guides/server/management-interface.adoc:25]: One thing we could add here, is that when a custom management path is not set, the custom context path via (--http-relative-path=auth) is still respected for the management endpoints, e.g. http://localhost:9000/auth/health and http://localhost:9000/auth/metrics.

**Comment by mabartos** [comment]
Resolved conflicts and ready for review again.

**Review by thomasdarimont — APPROVED** [review]
LGTM

**Comment by ahus1** [comment]
> What about ignoring the link for that test for now and rolling back it after the KC 25 release?

I agree, didn't see this this as the link will only start to work with KC25. Yes, let's add that link to the ignore list. 

BTW, there is a fix for the GitHub 429ers in main.

**Comment by mabartos** [comment]
@ahus1 @keycloak/cloud-native Are you okay with these changes?

**Comment by mabartos** [comment]
Rebased

**Review by ahus1 — DISMISSED** [review]
Thank you for these docs about the management interface!

File: docs/guides/server/management-interface.adoc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+includedOptions="http-management-* https-management-* legacy-observability-interface">
+
+The management interface allows accessing management endpoints via a different HTTP server than the primary one.
+It provides the possibility to hide endpoints like `/metrics` or `/health` from the outside world and, therefore, hardens the security.
+The most significant advantage might be seen in Kubernetes environments as the specific management port might not be exposed.
+
+== Management interface configuration
+
+The management interface is turned on by default, so management endpoints such as `/metrics`, and `/health` are exposed on the default management port `9000`.
+In order to change the port for the management interface, you can use the {project_name} option `http-management-port` or environment variable `KC_HTTP_MANAGEMENT_PORT`.
```
Comment (vmuzikar) [inline docs/guides/server/management-interface.adoc:17]: Just a nitpick. I would not mention the env vars. IMHO various config sources and formats are out of scope for this guide. If we do that, we might as well talk about format for keystore etc.

File: docs/guides/server/management-interface.adoc [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+NOTE: If you do not explicitly set the value for it, the value from the `http-relative-path` property is used. For instance,
+if you set the CLI option `--http-relative-path=/auth`, these endpoints are accessible on the `/auth/metrics` and `/auth/health` paths.
+
+== TLS support
+
+When the TLS is set for the default {project_name} server, the management interface will be accessible through HTTPS as well.
+The management interface can run only either on HTTP or HTTPS, not both as for the main server.
+
+Specific {project_name} management interface options with the prefix `https-management-*` were provided for setting different TLS parameters for the management HTTP server.
```
Comment (vmuzikar) [inline docs/guides/server/management-interface.adoc:35]:
```suggestion
Specific {project_name} management interface options with the prefix `https-management-*` were provided for setting different TLS parameters for the management HTTP server. Their function is similar to their counterparts for the main HTTP server, for details see <@links.server id="enabletls" />.
```

File: docs/guides/server/configuration-metrics.adoc [hunk 5]
```diff
@@ -18,10 +18,11 @@ It is possible to enable metrics using the build time option `metrics-enabled`:
 
 == Querying Metrics
 
-{project_name} exposes metrics at the following endpoint:
+{project_name} exposes metrics at the following endpoint on management port:
```
Comment (vmuzikar) [inline docs/guides/server/configuration-metrics.adoc:21]:
```suggestion
{project_name} exposes metrics at the following endpoint on the management interface at:
```

File: docs/guides/server/configuration-metrics.adoc [hunk 6]
```diff
@@ -18,10 +18,11 @@ It is possible to enable metrics using the build time option `metrics-enabled`:
...
 
 * `/metrics`
 
+For more information about the management port, see <@links.server id="management-interface" />.
```
Comment (vmuzikar) [inline docs/guides/server/configuration-metrics.adoc:25]:
```suggestion
For more information about the management interface, see <@links.server id="management-interface" />.
```

**Review by vmuzikar — COMMENTED** [review]
@mabartos Thank you for the docs! Added some suggestions.

**Comment by mabartos** [comment]
@ahus1 @vmuzikar Thanks! All your suggestions should be applied now.

**Review by vmuzikar — APPROVED** [review]

**Comment by mabartos** [comment]
Merging based on @vmuzikar's approval and @ahus1's previous approval before the minor polishing. The more we wait, the more conflicts will occur for the release notes, so I think it's ok for now. 

Thanks guys!

**Pull request merged**